Modules/xSharePoint/DSCResources/MSFT_xSPAntivirusSettings/MSFT_xSPAntivirusSettings.psm1

```diff
@@ -0,0 +1,113 @@
+        [parameter(Mandatory = $false)] [System.Boolean] $AllowDownloadInfected,
+        [parameter(Mandatory = $false)] [System.Boolean] $AttemptToClean,
+        [parameter(Mandatory = $false)] [System.UInt16] $TimeoutDuration,
+        [parameter(Mandatory = $false)] [System.UInt16] $NumberOfThreads,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    Write-Verbose -Message "Getting antivirus configuration settings"
+
+    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+        
+        [System.Reflection.Assembly]::LoadWithPartialName("Microsoft.SharePoint") | Out-Null
+
+        # Get a reference to the Administration WebService
+        $admService = [Microsoft.SharePoint.Administration.SPWebService]::ContentService
+
+        if ($null -eq $admService) { return $null }
+        
+        return @{
+            # Set the antivirus settings
+            AllowDownloadInfected = $admService.AntivirusSettings.AllowDownload
+            ScanOnDownload = $admService.AntivirusSettings.DownloadScanEnabled
+            ScanOnUpload = $admService.AntivirusSettings.UploadScanEnabled
+            AttemptToClean = $admService.AntivirusSettings.CleaningEnabled
+            NumberOfThreads = $admService.AntivirusSettings.NumberOfThreads
+            TimeoutDuration = $admService.AntivirusSettings.Timeout.TotalSeconds
+            InstallAccount = $params.InstallAccount
+        }
+    }
+
+    return $result
+}
+
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [parameter(Mandatory = $true)] [System.Boolean] $ScanOnDownload,
+        [parameter(Mandatory = $false)] [System.Boolean] $ScanOnUpload,
+        [parameter(Mandatory = $false)] [System.Boolean] $AllowDownloadInfected,
+        [parameter(Mandatory = $false)] [System.Boolean] $AttemptToClean,
+        [parameter(Mandatory = $false)] [System.UInt16] $TimeoutDuration,
+        [parameter(Mandatory = $false)] [System.UInt16] $NumberOfThreads,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    Write-Verbose -Message "Setting antivirus configuration settings"
+
+    Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+        
+        Write-Verbose -Message "Start update"
+        # Load the SharePoint Assembly, using old style for backward compatibility with V1
+        [System.Reflection.Assembly]::LoadWithPartialName("Microsoft.SharePoint") | Out-Null
+
+        # Get a reference to the Administration WebService
+        $admService = [Microsoft.SharePoint.Administration.SPWebService]::ContentService
+
+        # Set the antivirus settings
+        if ($params.ContainsKey("AllowDownloadInfected")) { 
+            Write-Verbose -Message "Setting Allow Download"
```

Comment:
Just performed a small test. When running the script below, I do not see any verbose output:
$credential = Get-Credential
$server = "<server>"

$session = New-PSSession -ComputerName $server -Credential $Credential -Authentication CredSSP -ErrorAction Continue
$result = Invoke-Command -Session $session -ScriptBlock {Write-Verbose "This is a test"} -ArgumentList $test -Verbose
$result

Also found many articles on the Internet stating that the Verbose output in Invoke-Command will not work that well. Will test with the DSC resource as well.